### ISSUE TITLE
aws-s3-deployment: SourceMarkers contains duplicates

### ISSUE THREAD

**Issue by yeDor** [aws/aws-cdk#21504]
### Describe the bug

if same deploy-time value used more then once, produced template conteans duplicates of corresponding marker.
does not affects the functionality, 

### Expected Behavior

each distinct marker generated only once

### Current Behavior

duplicates are present:
```
...
    "SourceMarkers": [
     {
      "<<marker:0xbaba:0>>": {
       "Fn::GetAtt": [
        "testBucketDF4D7D1A",
        "DomainName"
       ]
      },
      "<<marker:0xbaba:1>>": {
       "Fn::GetAtt": [
        "testBucketDF4D7D1A",
        "RegionalDomainName"
       ]
      }
     },
     {
      "<<marker:0xbaba:0>>": {
       "Fn::GetAtt": [
        "testBucketDF4D7D1A",
        "DomainName"
       ]
      },
      "<<marker:0xbaba:1>>": {
       "Fn::GetAtt": [
        "testBucketDF4D7D1A",
        "RegionalDomainName"
       ]
      }
     }
    ],
...
```

### Reproduction Steps

```
@Getter
@ToString
public class Stack2 extends Stack {

  private final BucketDeployment dwcWebDeployment;

  @SneakyThrows
  @lombok.Builder
  Stack2(
    @NonNull Construct scope,
    @NonNull StackProps props,
    @NonNull Stack1 stack1
  ) {
    super(scope, "Stack2", props);

    Bucket testBucket = Bucket.Builder.create(this, "testBucket").build();

    WebConfig webConfig = WebConfig.builder()
                                   .hostUrl(testBucket.getBucketDomainName())
                                   .s3HostUrl(testBucket.getBucketRegionalDomainName()  )
                                   .build();

    dwcWebDeployment = BucketDeployment.Builder.create(this, "deployment")
                                               .destinationBucket(testBucket)
                                               .logRetention(RetentionDays.ONE_MONTH)
                                               .sources(asList(
                                                 Source.jsonData("config.json", webConfig),
                                                 Source.jsonData("config2.json", webConfig)
                                               ))
                                               .build();
  }

  @lombok.Builder
  @Data
  @AllArgsConstructor
  @NoArgsConstructor
  public static class WebConfig {

    @NonNull String hostUrl;
    @NonNull String s3HostUrl;
  }

}

```

### CDK CLI Version

2.27.0 (build 8e89048)

### Node.js Version

v18.7.0

### OS

Windows 10

### Language

Java
